841.py

```diff
@@ -0,0 +1,38 @@
+class Solution:
+    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
+        visited = set()
+
+        def dfs(room: int):
+            if room in visited:
+                return False
+
+            visited.add(room)
+
+            if len(visited) == len(rooms):
```

Comment:
When designing or writing recursive functions, we should always ask ourselves what the base case is (well, sometimes there are more than one base case) and what the recursive cases are.

Here, `len(visited) == len(rooms)` is a condition related to the `canVisitAllRooms` problem, not the problem this `dfs()` recursive function is supposed to solve. Essentially you are mixing up or coupling two problems into one function, which is not a good practice. And when the problem gets more complex than this simple DFS problem, you will get lost in the middle very easily.